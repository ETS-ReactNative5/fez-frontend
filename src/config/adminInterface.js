// import React from 'react';
import { PUBLICATION_TYPE_JOURNAL_ARTICLE } from 'config/general';
import Immutable from 'immutable';

import { validation } from 'config';
import locale from 'locale/components';
import { default as formLocale } from 'locale/publicationForm';

import { ListEditorField } from 'modules/SharedComponents/Toolbox/ListEditor';
import { LinkInfoListEditorField } from 'modules/SharedComponents/Toolbox/ListEditor';
import { LanguageField } from 'modules/SharedComponents/Toolbox/LanguageField';
import { WoSDocTypesField } from 'modules/SharedComponents/Toolbox/WoSDocTypesField';
import { ScopusDocTypesField } from 'modules/SharedComponents/Toolbox/ScopusDocTypesField';
import { PubmedDocTypesField } from 'modules/SharedComponents/Toolbox/PubmedDocTypesField';
import { TextField as GenericTextField } from 'modules/SharedComponents/Toolbox/TextField';
import { RichEditorField } from 'modules/SharedComponents/RichEditor';
import { CollectionField } from 'modules/SharedComponents/LookupFields';
import { PublicationSubtypeField } from 'modules/SharedComponents/PublicationSubtype';
import { ContributorsEditorField } from 'modules/SharedComponents/ContributorsEditor';

export const fieldConfig = {
    rek_title: {
        component: RichEditorField,
        componentProps: {
            name: 'bibliographicSection.rek_title',
            title: 'Formatted title',
            titleProps: {
                variant: 'caption',
                style: {
                    opacity: 0.666
                }
            },
            height: 100,
            format: (value) => Immutable.Map(value),
            validate: [validation.required]
        }
    },
    rek_herdc_notes: {
        component: RichEditorField,
        componentProps: {
            name: 'adminSection.rek_herdc_notes',
            title: 'HERDC notes',
            disabled: true,
            titleProps: {
                variant: 'caption',
                style: {
                    opacity: 0.666
                }
            },
            height: 100,
            format: (value) => Immutable.Map(value)
        }
    },
    fez_internal_notes: {
        component: RichEditorField,
        componentProps: {
            name: 'adminSection.fez_internal_notes.ain_detail',
            title: 'Internal notes',
            titleProps: {
                variant: 'caption',
                style: {
                    opacity: 0.666
                }
            },
            height: 100,
            format: (value) => Immutable.Map(value)
        }
    },
    fez_record_search_key_isi_loc: {
        component: GenericTextField,
        componentProps: {
            name: 'identifiersSection.fez_record_search_key_isi_loc.rek_isi_loc',
            fullWidth: true,
            label: 'WoS ID',
            placeholder: ''
        }
    },
    fez_record_search_key_scopus_id: {
        component: GenericTextField,
        componentProps: {
            name: 'identifiersSection.fez_record_search_key_scopus_id.rek_scopus_id',
            fullWidth: true,
            label: 'Scopus ID',
            placeholder: ''
        }
    },
    fez_record_search_key_pubmed_id: {
        component: GenericTextField,
        componentProps: {
            name: 'identifiersSection.fez_record_search_key_pubmed_id.rek_pubmed_id',
            fullWidth: true,
            label: 'PubMed ID',
            placeholder: ''
        }
    },
    rek_wok_doc_type: {
        component: WoSDocTypesField,
        componentProps: {
            name: 'identifiersSection.rek_wok_doc_type',
            label: 'WoS Doc Type',
            placeholder: ''
        }
    },
    rek_scopus_doc_type: {
        component: ScopusDocTypesField,
        componentProps: {
            name: 'identifiersSection.rek_scopus_doc_type',
            label: 'Scopus Doc Type',
            placeholder: ''
        }
    },
    rek_pubmed_doc_type: {
        component: PubmedDocTypesField,
        componentProps: {
            name: 'identifiersSection.rek_pubmed_doc_type',
            label: 'PubMed Doc Type',
            placeholder: ''
        }
    },
    links: {
        component: LinkInfoListEditorField,
        componentProps: {
            name: 'identifiersSection.links',
            label: 'Link',
            placeholder: '',
            locale: locale.components.linkListForm.field
        }
    },
    rek_description: {
        component: RichEditorField,
        componentProps: {
            name: 'bibliographicSection.rek_description',
            title: 'Formatted abstract',
            titleProps: {
                variant: 'caption',
                style: {
                    opacity: 0.666
                }
            },
            height: 100,
            format: (value) => Immutable.Map(value),
            validate: [validation.required]
        }
    },
    rek_date: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliogrphicSection.rek_date',
            label: 'Publication date',
            placeholder: 'Date of publication',
            required: true,
            fullWidth: true
        }
    },
    collections: {
        component: CollectionField,
        componentProps: {
            label: 'Collection',
            placeholder: 'Begin typing to select and add collection(s)',
            name: 'bibliographicSection.collections'
        }
    },
    rek_subtype: {
        component: PublicationSubtypeField,
        componentProps: {
            name: 'bibliographicSection.rek_subtype',
            label: 'eSpace subtype',
            required: true,
            placeholder: ''
        }
    },
    languages: {
        component: LanguageField,
        componentProps: {
            name: 'bibliographicSection.languages',
            label: 'Language',
            required: true,
            placeholder: '',
            multiple: true
        }
    },
    fez_record_search_key_journal_name: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_journal_name.rek_journal_name',
            fullWidth: true,
            label: 'Journal name',
            placeholder: '',
            required: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_doi: {
        component: GenericTextField,
        componentProps: {
            name: 'identifiersSection.fez_record_search_key_doi.rek_doi',
            fullWidth: true,
            label: 'DOI',
            placeholder: ''
        }
    },
    fez_record_search_key_place_of_publication: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_place_of_publication.rek_place_of_publication',
            fullWidth: true,
            label: 'Place of publication',
            placeholder: '',
            required: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_publisher: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_publisher.rek_publisher',
            fullWidth: true,
            label: 'Publisher',
            placeholder: '',
            required: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_volume_number: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_volume_number.rek_volume_number',
            fullWidth: true,
            label: 'Volume',
            placeholder: '',
            required: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_issue_number: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_issue_number.rek_issue_number',
            fullWidth: true,
            label: 'Issue',
            placeholder: '',
            required: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_article_number: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_article_number.rek_article_number',
            fullWidth: true,
            label: 'Article Number',
            placeholder: ''
        }
    },
    fez_record_search_key_start_page: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_start_page.rek_start_page',
            fullWidth: true,
            label: 'Start page',
            placeholder: '',
            required: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_end_page: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_end_page.rek_end_page',
            fullWidth: true,
            label: 'End page',
            placeholder: '',
            required: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_oa_embargo_days: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_oa_embargo_days.rek_oa_embargo_days',
            fullWidth: true,
            label: 'DOI Embargo days',
            placeholder: '',
            required: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_keywords: {
        component: ListEditorField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_keywords',
            required: true,
            validate: [validation.requiredList],
            maxInputLength: 111,
            searchKey: {
                value: 'rek_keywords',
                order: 'rek_keywords_order'
            },
            locale: locale.components.keywordsForm.field
        }
    },
    fez_record_search_key_issn: {
        component: ListEditorField,
        componentProps: {
            remindToAdd: true,
            name: 'bibliographicSection.fez_record_search_key_issn',
            isValid: validation.isValidIssn,
            searchKey: {
                value: 'rek_issn',
                order: 'rek_issn_order'
            },
            locale: locale.components.issnForm.field
        }
    },
    fez_record_search_key_isbn: {
        component: ListEditorField,
        componentProps: {
            remindToAdd: true,
            name: 'bibliographicSection.fez_record_search_key_isbn',
            isValid: validation.isValidIsbn,
            searchKey: {
                value: 'rek_isbn',
                order: 'rek_isbn_order'
            },
            locale: locale.components.isbnForm.field
        }
    },
    fez_record_search_key_ismn: {
        component: ListEditorField,
        componentProps: {
            remindToAdd: true,
            name: 'bibliographicSection.fez_record_search_key_ismn',
            isValid: validation.isValidIsmn,
            searchKey: {
                value: 'rek_ismn',
                order: 'rek_ismn_order'
            },
            locale: locale.components.ismnForm.field
        }
    },
    fez_record_search_key_total_pages: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_total_pages.rek_total_pages',
            fullWidth: true,
            label: 'Total pages',
            placeholder: '',
            required: true,
            validate: [validation.required]
        }
    },
    subjects: {
        component: '',
        componentProps: {
            name: ''
        }
    },
    languageOfJournalName: {
        component: LanguageField,
        componentProps: {
            name: 'bibliographicSection.languageOfJournalName',
            label: 'Language of journal name',
            placeholder: '',
            required: true,
            multiple: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_native_script_journal_name: {
        component: GenericTextField,
        componentProps: {
            name:
				'bibliographicSection.fez_record_search_key_native_script_journal_name.rek_native_script_journal_name',
            label: 'Native script journal name',
            placeholder: '',
            fullWidth: true
        }
    },
    fez_record_search_key_roman_script_journal_name: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_roman_script_journal_name.rek_roman_script_journal_name',
            label: 'Roman script journal name',
            placeholder: '',
            fullWidth: true
        }
    },
    fez_record_search_key_translated_journal_name: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_translated_journal_name.rek_translated_journal_name',
            label: 'Translated journal name',
            placeholder: '',
            fullWidth: true
        }
    },
    languageOfTitle: {
        component: LanguageField,
        componentProps: {
            name: 'bibliographicSection.languageOfTitle',
            label: 'Language of title',
            placeholder: '',
            required: true,
            multiple: true,
            validate: [validation.required]
        }
    },
    fez_record_search_key_native_script_title: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_native_script_title.rek_native_script_title',
            label: 'Native script title',
            placeholder: '',
            fullWidth: true
        }
    },
    fez_record_search_key_roman_script_title: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_roman_script_title.rek_roman_script_title',
            label: 'Roman script title',
            placeholder: '',
            fullWidth: true
        }
    },
    fez_record_search_key_translated_title: {
        component: GenericTextField,
        componentProps: {
            name: 'bibliographicSection.fez_record_search_key_translated_title.rek_translated_title',
            label: 'Translated title',
            placeholder: '',
            fullWidth: true
        }
    },
    authors: {
        component: ContributorsEditorField,
        componentProps: {
            name: 'authorsSection.authors',
            showIdentifierLookup: true,
            showContributorAssignment: true,
            locale: formLocale.journalArticle.authors.field,
            validate: [validation.authorRequired]
        }
    }
};

export const adminInterfaceConfig = {
    [PUBLICATION_TYPE_JOURNAL_ARTICLE]: {
        admin: () => [
            {
                groups: [
                    ['fez_internal_notes'],
                    ['rek_herdc_notes']
                    // ['fez_record_search_key_retracted']
                ]
            }
        ],
        identifiers: () => [
            {
                title: 'Manager identifiers',
                groups: [
                    ['fez_record_search_key_doi'],
                    ['fez_record_search_key_isi_loc', 'rek_wok_doc_type'],
                    ['fez_record_search_key_scopus_id', 'rek_scopus_doc_type'],
                    ['fez_record_search_key_pubmed_id', 'rek_pubmed_doc_type']
                ]
            },
            {
                title: 'Manage links',
                groups: [['links']]
            }
        ],
        bibliographic: (isLote = false, isNtro = false) => [
            {
                title: 'Title',
                groups: [
                    ['rek_title'],
                    ...(isLote
                        ? [
                            ['languageOfTitle'],
                            ['fez_record_search_key_native_script_title'],
                            ['fez_record_search_key_translated_title'],
                            ['fez_record_search_key_roman_script_title']
						  ]
                        : []),
                    ['languages']
                ]
            },
            {
                title: 'Journal name',
                groups: [
                    ['fez_record_search_key_journal_name'],
                    ...(isLote
                        ? [
                            ['languageOfJournalName'],
                            ['fez_record_search_key_native_script_journal_name'],
                            ['fez_record_search_key_translated_journal_name'],
                            ['fez_record_search_key_roman_script_journal_name']
						  ]
                        : []),
                    ['rek_subtype']
                    // ['fez_record_search_key_doi', 'fez_record_search_key_oa_embargo_days']
                ]
            },
            {
                title: 'ISSN',
                groups: [['fez_record_search_key_issn']]
            },
            {
                title: 'ISBN',
                groups: [['fez_record_search_key_isbn']]
            },
            ...(isNtro
                ? [
                    {
                        title: 'ISMN',
                        groups: [['fez_record_search_key_ismn']]
                    }
				  ]
                : []),
            {
                title: 'Bibliographic',
                groups: [
                    ['fez_record_search_key_place_of_publication', 'fez_record_search_key_publisher'],
                    [
                        'fez_record_search_key_volume_number',
                        'fez_record_search_key_issue_number',
                        'fez_record_search_key_article_number'
                    ],
                    [
                        'fez_record_search_key_start_page',
                        'fez_record_search_key_end_page',
                        'fez_record_search_key_total_pages'
                    ],
                    ['rek_date'],
                    // ['rek_date_available'],
                    // ['rek_refereed_source'],
                    // ['fez_record_search_key_succeeds'],
                    ['rek_description']
                ]
            },
            {
                title: 'Keywords',
                groups: [['fez_record_search_key_keywords']]
            },
            {
                title: 'Subject',
                groups: [
                    // ['fez_record_search_key_for_codes']
                ]
            }
        ],
        authors: () => [
            {
                title: 'Authors',
                groups: [['authors']]
            }
        ]
    }
};

export const valueExtractor = {
    rek_title: {
        getValue: (record) => ({
            plainText: record.rek_title,
            htmlText: record.rek_formatted_title || record.rek_title
        })
    },
    rek_description: {
        getValue: (record) => ({
            plainText: record.rek_description,
            htmlText: record.rek_formatted_abstract || record.rek_description
        })
    },
    rek_date: {
        getValue: (record) => record.rek_date
    },
    rek_subtype: {
        getValue: (record) => record.rek_subtype
    },
    languages: {
        getValue: (record) => record.fez_record_search_key_language.map((language) => language.rek_language)
    },
    fez_record_search_key_journal_name: {
        getValue: (record) => ({ ...record.fez_record_search_key_journal_name })
    },
    fez_record_search_key_place_of_publication: {
        getValue: (record) => ({
            ...record.fez_record_search_key_place_of_publication
        })
    },
    fez_record_search_key_publisher: {
        getValue: (record) => ({ ...record.fez_record_search_key_publisher })
    },
    fez_record_search_key_volume_number: {
        getValue: (record) => ({ ...record.fez_record_search_key_volume_number })
    },
    fez_record_search_key_issue_number: {
        getValue: (record) => ({ ...record.fez_record_search_key_issue_number })
    },
    fez_record_search_key_article_number: {
        getValue: (record) => ({ ...record.fez_record_search_key_article_number })
    },
    fez_record_search_key_start_page: {
        getValue: (record) => ({ ...record.fez_record_search_key_start_page })
    },
    fez_record_search_key_end_page: {
        getValue: (record) => ({ ...record.fez_record_search_key_end_page })
    },
    fez_record_search_key_oa_embargo_days: {
        getValue: (record) => ({
            ...record.fez_record_search_key_oa_embargo_days
        })
    },
    fez_record_search_key_total_pages: {
        getValue: (record) => ({
            ...record.fez_record_search_key_total_pages
        })
    },
    collections: {
        getValue: (record) =>
            record.fez_record_search_key_ismemberof.map((collection) => ({
                id: collection.rek_ismemberof,
                value: collection.rek_ismemberof_lookup
            }))
    },
    fez_record_search_key_keywords: {
        getValue: (record) => [...record.fez_record_search_key_keywords]
    },
    fez_record_search_key_issn: {
        getValue: (record) => [...record.fez_record_search_key_issn]
    },
    fez_record_search_key_isbn: {
        getValue: (record) => [...record.fez_record_search_key_isbn]
    },
    fez_record_search_key_ismn: {
        getValue: (record) => [...record.fez_record_search_key_ismn]
    },
    subjects: {
        getValue: (record) =>
            record.fez_record_search_key_subject.map((subject) => ({
                rek_value: {
                    key: subject.rek_subject,
                    value: subject.rek_subject_lookup
                },
                rek_order: subject.rek_subject_order
            }))
    },
    languageOfJournalName: {
        getValue: (record) =>
            record.fez_record_search_key_language_of_journal_name.map(
                (language) => language.rek_language_of_journal_name
            )
    },
    fez_record_search_key_native_script_journal_name: {
        getValue: (record) =>
            (record.fez_record_search_key_native_script_journal_name || {}).rek_native_script_journal_name
    },
    fez_record_search_key_roman_script_journal_name: {
        getValue: (record) =>
            (record.fez_record_search_key_roman_script_journal_name || {}).rek_roman_script_journal_name
    },
    fez_record_search_key_translated_journal_name: {
        getValue: (record) => (record.fez_record_search_key_translated_journal_name || {}).rek_translated_journal_name
    },
    languageOfTitle: {
        getValue: (record) =>
            record.fez_record_search_key_language_of_title.map((language) => language.rek_language_of_title)
    },
    fez_record_search_key_native_script_title: {
        getValue: (record) => (record.fez_record_search_key_native_script_title || {}).rek_native_script_title
    },
    fez_record_search_key_roman_script_title: {
        getValue: (record) => (record.fez_record_search_key_roman_script_title || {}).rek_roman_script_title
    },
    fez_record_search_key_translated_title: {
        getValue: (record) => (record.fez_record_search_key_translated_title || {}).rek_translated_title
    },
    fez_record_search_key_doi: {
        getValue: (record) => ({ ...record.fez_record_search_key_doi })
    },
    fez_record_search_key_isi_loc: {
        getValue: (record) => ({ ...record.fez_record_search_key_isi_loc })
    },
    fez_record_search_key_scopus_id: {
        getValue: (record) => ({ ...record.fez_record_search_key_scopus_id })
    },
    fez_record_search_key_pubmed_id: {
        getValue: (record) => ({ ...record.fez_record_search_key_pubmed_id })
    },
    rek_wok_doc_type: {
        getValue: (record) => record.rek_wok_doc_type
    },
    rek_scopus_doc_type: {
        getValue: (record) => record.rek_scopus_doc_type
    },
    rek_pubmed_doc_type: {
        getValue: (record) => record.rek_pubmed_doc_type
    },
    links: {
        getValue: (record) =>
            (record.fez_record_search_key_link || []).map((link) => ({
                rek_order: link.rek_link_order,
                rek_value: {
                    key: link.rek_link,
                    value: record.fez_record_search_key_link_description
                        .filter((description) => description.rek_link_description_order === link.rek_link_order)
                        .reduce((pv, cv) => cv.rek_link_description, '')
                }
            }))
    },
    authors: {
        getValue: (record) => {
            const authors = (record.fez_record_search_key_author || []).reduce(
                (authorsObject, author) => ({
                    ...authorsObject,
                    [author.rek_author_order]: author
                }),
                {}
            );

            const authorIds = (record.fez_record_search_key_author_id || []).reduce(
                (authorIdsObject, authorId) => ({
                    ...authorIdsObject,
                    [authorId.rek_author_id_order]: authorId
                }),
                {}
            );

            return (record.fez_record_search_key_author || []).map(({ rek_author_order: order }) => {
                return {
                    nameAsPublished: authors[order].rek_author,
                    creatorRole: '',
                    uqIdentifier: `${authorIds[order].rek_author_id}` || '',
                    authorId: authorIds[order].rek_author_id
                };
            });
        }
    }
};
