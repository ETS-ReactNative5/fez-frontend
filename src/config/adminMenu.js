// menuItems to display
export const menuItems = email => [
    {
        linkTo: '/admin',
        primaryText: 'My dashboard',
        secondaryText: email
    },
    {
        divider: true
    }
];

