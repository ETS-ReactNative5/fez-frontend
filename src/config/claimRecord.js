export const claimRecordConfig = {
    hideContributorLinking: [
        130, // Book Chapter
        177, // Conference Paper
        179  // Journal Article
    ]
};
