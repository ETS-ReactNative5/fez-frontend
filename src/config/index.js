export {
    API_URL,
    APP_URL,
    AUTH_URL_LOGIN,
    AUTH_URL_LOGOUT,
    SESSION_COOKIE_NAME,
    TOKEN_NAME,
    ORCID_BASE_URL,
    ORCID_CLIENT_ID,
    ORCID_AUTHORIZATION_URL,
    FACET_TYPE_FILTER,
    FACET_TYPE_RANGE,
    FEZ_URL
} from './general';

export {api, generateCancelToken, cache} from './axios';
export {default as theme} from './theme';
export * as validation from './validation';
export * as general from './general';
export {publicationTypes} from './general';
export * as routes from './routes';
