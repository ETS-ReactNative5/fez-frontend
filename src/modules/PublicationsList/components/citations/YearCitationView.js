import React from 'react';
import PropTypes from 'prop-types';

const YearCitationView = ({publication}) => {
    console.log(publication.rek_date);
    if (!publication.rek_date) return (<span className="citationYear empty" />);

    const year = !isNaN(new Date(publication.rek_date))
        ? new Date(publication.rek_date)
        : new Date(publication.rek_date.replace(/-/g, '/'));

    if (!year || isNaN(year.getFullYear())) return (<span className="citationYear empty" />);
    return (<span className="citationYear"> ({year.getFullYear()}) </span>);
};

YearCitationView.propTypes = {
    publication: PropTypes.object.isRequired
};

export default YearCitationView;
