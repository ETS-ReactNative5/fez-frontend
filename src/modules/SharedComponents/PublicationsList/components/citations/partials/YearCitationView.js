import React from 'react';
import PropTypes from 'prop-types';

const YearCitationView = ({date}) => {
    if (!date) return (<span className="citationYear empty" />);

    const year = !isNaN(new Date(date))
        ? new Date(date)
        : new Date(date.replace(/-/g, '/'));

    if (!year || isNaN(year.getFullYear())) return (<span className="citationYear empty" />);
    return (<span className="citationYear">({year.getFullYear()}).</span>);
};

YearCitationView.propTypes = {
    date: PropTypes.string
};

export default YearCitationView;
