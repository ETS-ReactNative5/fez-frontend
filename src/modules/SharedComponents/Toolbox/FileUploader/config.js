export const CLOSED_ACCESS_ID = 8;
export const OPEN_ACCESS_ID = 9;
export const SIZE_BASE = 1000;

export const DEFAULT_FILE_UPLOAD_LIMIT = 10;
export const DEFAULT_MAX_FILE_SIZE = 8;

export const SIZE_UNIT_B = 'B';
export const SIZE_UNIT_K = 'K';
export const SIZE_UNIT_M = 'M';
export const SIZE_UNIT_G = 'G';

export const FILE_META_KEY_ACCESS_CONDITION = 'access_condition_id';
export const FILE_META_KEY_EMBARGO_DATE = 'date';

export const SIZE_UNITS = [SIZE_UNIT_B, SIZE_UNIT_K, SIZE_UNIT_M, SIZE_UNIT_G];

export const FILE_NAME_RESTRICTION = /^(?=^\S*$)(?=^[a-z\d\-_]+(\.\d{3}|\.r\d{2,3}|\.part\d{1,3})?\.[^\.]+$)(?=.{1,45}$)(?!(web_|preview_|thumbnail_|stream_|fezacml_|presmd_|\d))[a-z\d\-_\.]+/;

/**
 * Note: ext -> mimetype is a 1-n relation, but we only care about one value
 */
export const MIME_TYPE_WHITELIST = {
    pdf: 'application/pdf',
    jpe: 'image/jpeg',
    jpeg: 'image/jpeg',
    jpg: 'image/jpeg',
    gif: 'image/gif',
    tif: 'image/tiff',
    tiff: 'image/tiff',
    png: 'image/png',
    zip: 'application/zip',
    tar: 'application/x-tar',
    gz: 'application/gzip',
    '7z': 'application/x-7z-compressed',
    rar: 'application/x-rar-compressed',
    mk3d: 'video/x-matroska',
    mks: 'video/x-matroska',
    mkv: 'video/x-matroska',
    qt: 'video/quicktime',
    mov: 'video/quicktime',
    mp4: 'video/mp4',
    mp4v: 'video/mp4',
    mp3: 'audio/mp3',
    wav: 'audio/wav',
    mpe: 'video/mpeg',
    mpeg: 'video/mpeg',
    mpg: 'video/mpeg',
    m1v: 'video/mpeg',
    m2v: 'video/mpeg',
    wmv: 'video/x-ms-wmv',
    wma: 'audio/x-ms-wma',
    avi: 'video/x-msvideo',
    m4a: 'audio/x-m4a',
    csv: 'text/csv',
    xla: 'application/vnd.ms-excel',
    xlc: 'application/vnd.ms-excel',
    xlm: 'application/vnd.ms-excel',
    xls: 'application/vnd.ms-excel',
    xlt: 'application/vnd.ms-excel',
    xlw: 'application/vnd.ms-excel',
    xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
    ods: 'application/vnd.oasis.opendocument.spreadsheet',
    gsheet: 'application/vnd.google-apps.spreadsheet',
};
