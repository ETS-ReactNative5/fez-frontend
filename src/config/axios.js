import axios from 'axios';
import Cookies from 'js-cookie';
import {setupCache} from 'axios-cache-adapter';
import {API_URL, SESSION_COOKIE_NAME, TOKEN_NAME} from './general';
import {store} from 'config/store';
import {logout} from 'actions/account';
import {locale} from 'locale';

export const cache = setupCache({
    maxAge: 15 * 60 * 1000,
    key: (request) => {
        return request.url + JSON.stringify(request.params);
    },
    exclude: {
        query: false,
        paths: [
            'external/records/search',
            'records/search?rule=',
            'records/search?title=',
            'records/search?doi=',
            'records/search?id=pmid:',
            'orcid'
        ]
    },
});

export const api = axios.create({
    baseURL: API_URL,
    adapter: process.env.NODE_ENV === 'test' ? undefined : cache.adapter
});

// need to generate a new token for each request otherwise if you try a new request with the old token,
// axios will appear to cancel your request automatically
export const generateCancelToken = () => {
    const CancelToken = axios.CancelToken;
    return CancelToken.source();
};

api.defaults.headers.common[TOKEN_NAME] = Cookies.get(SESSION_COOKIE_NAME);
api.isCancel = axios.isCancel; // needed for cancelling requests and the instance created does not have this method

let isGet = null;
api.interceptors.request.use(request => {
    isGet = request.method === 'get';
    return request;
});

api.interceptors.response.use(response => {
    if (!isGet) {
        return cache.store.clear().then(() => Promise.resolve(response.data));
    }
    return Promise.resolve(response.data);
}, error => {
    let errorMessage = locale.global.errorMessages.generic;

    if (error.response && error.response.status === 403) {
        errorMessage = locale.global.errorMessages.sessionExpired;
        if (process.env.NODE_ENV === 'test') {
            global.mockActionsStore.dispatch(logout());
        } else {
            store.dispatch(logout());
        }
    } else if (error.response && error.response.status === 404) {
         errorMessage = locale.global.errorMessages.notFound;
    }

    const errorDetails = {
        status: error.response ? error.response.status : null,
        message: errorMessage
    };

    return Promise.reject(errorDetails);
});
