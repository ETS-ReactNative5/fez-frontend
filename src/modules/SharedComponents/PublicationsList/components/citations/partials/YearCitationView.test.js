jest.dontMock('./YearCitationView');

import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import YearCitationView from './YearCitationView';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import PropTypes from 'prop-types';
import injectTapEventPlugin from 'react-tap-event-plugin';
import {locale} from 'config';


function setup({date, isShallow = true}) {
    const props = {
        date: date, // : PropTypes.object.isRequired,
    };

    if(isShallow) {
        return shallow(<YearCitationView {...props} />);
    }

    return mount(<YearCitationView {...props} />, {
        context: {
            muiTheme: getMuiTheme()
        },
        childContextTypes: {
            muiTheme: PropTypes.object.isRequired
        }
    });
}

beforeAll(() => {
    injectTapEventPlugin();
});

describe('YearCitationView test ', () => {
    it('should render component with no date', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component with date not in TZ format', () => {
        const testObject = {
            'rek_date': '2010-08-01 00:00:00'
        };
        const wrapper = setup({ date: testObject.rek_date });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component with date in TZ format', () => {
        const testObject = {
            'rek_date': '2017-07-01T00:00:00Z'
        };
        const wrapper = setup({ date: testObject.rek_date });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component with date invalid date format', () => {
        const testObject = {
            'rek_date': 'BLA BLA BLA'
        };
        const wrapper = setup({ date: testObject.rek_date });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});
