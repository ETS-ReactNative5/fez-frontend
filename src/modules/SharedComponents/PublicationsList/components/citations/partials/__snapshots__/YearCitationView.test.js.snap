// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearCitationView test  should render component with TZ date and custom prefix and suffix 1`] = `
<YearCitationView
  date="2017-07-01T00:00:00Z"
  prefix="("
  suffix=")."
>
  <span
    className="citationYear"
  >
    (
    2017
    ).
  </span>
</YearCitationView>
`;

exports[`YearCitationView test  should render component with date in TZ format 1`] = `
<YearCitationView
  date="2017-07-01T00:00:00Z"
  prefix="("
  suffix=")."
>
  <span
    className="citationYear"
  >
    (
    2017
    ).
  </span>
</YearCitationView>
`;

exports[`YearCitationView test  should render component with date invalid date format 1`] = `
<YearCitationView
  date="BLA BLA BLA"
  prefix="("
  suffix=")."
>
  <span
    className="citationYear empty"
  />
</YearCitationView>
`;

exports[`YearCitationView test  should render component with date not in TZ format 1`] = `
<YearCitationView
  date="2010-08-01 00:00:00"
  prefix="("
  suffix=")."
>
  <span
    className="citationYear"
  >
    (
    2010
    ).
  </span>
</YearCitationView>
`;

exports[`YearCitationView test  should render component with no date 1`] = `
<YearCitationView
  prefix="("
  suffix=")."
>
  <span
    className="citationYear empty"
  />
</YearCitationView>
`;
